src/content_sao.cpp

```diff
@@ -1336,11 +1336,13 @@ void PlayerSAO::setHP(s32 hp, const PlayerHPChangeReason &reason)
 
 	hp = rangelim(hp, 0, m_prop.hp_max);
 
-	s32 hp_change = m_env->getScriptIface()->on_player_hpchange(this, hp - oldhp, reason);
-	if (hp_change == 0)
-		return;
+	if (m_hp != hp) {
```

Comment:
why does the `rangelim` code use oldhp, but this one uses m_hp?

Reply:
There's no real reason for it - they're both the same anyway. I could make this check against `oldhp` instead if you want. I could also move the oldhp declaration below this if block to eliminate this confusion.

Reply:
For consistency I'd use `oldhp` in both cases

Reply:
Alright, done. :)